### DISCUSSION TITLE
Wikilinks and hierarchical project structure

### DISCUSSION THREAD

**Discussion by Martien** [observablehq/framework#1838] [Q&A] [answered]
Torn between two lovers…

- framework says, [Use folders to organize your pages.](https://observablehq.com/framework/project-structure#src) – I am all in favour, especially when having many pages: oversight;
- now take [Configuration#pages](https://observablehq.com/framework/config#pages) as an example; and
- combine this with [markdown-it-wikilinks](https://github.com/observablehq/framework/tree/82922bd3d720e39f093fed8254194e97d7ff4821/examples/markdown-it-wikilinks).

That is, having this:
```
export default {
  pages: [
    {
      name: "Section 1",
      path: "/s01/",
      pages: [
        {name: "Page 1", path: "/s01/page1"},
        {name: "Page 2", path: "/s01/page2"}
      ]
    },
    {
      name: "Section 2",
      open: false,
      pages: [
        {name: "Page 3", path: "/s02/page3"},
        {name: "Page 4", path: "/s02/page4"}
      ]
    }
  ]
}
```
and using `[[wiki links]]`:
- in “Page **1**” refers to `/s01/wiki-links`;
- in “Page **3**“ refers to `/s02/wiki-links`.

This makes sense when ‘realms’ like `Section 1` and `Section 2` may have individual versions of identically named pages.

However, Wikipedia in general has a flat structure: one level deep and unique page names, with disambiguation pages coming to the rescue of resolving naming conflicts.

So, the questions is: how to best combine a predominantly flat name space for Wikilinks – so `[[wiki link]]` always refers uniquely to the same page `/wiki-link` while also allowing for a hierarchical source structure?

To support a flat `[[wiki link]]` space, the most straightforward way is to put _**all**_ sources in the root `/src`, but this clashes with using folders to organise pages.

If I miss some crucial information or insight, please enlighten me.

**Comment by mbostock** [comment, marked as answer]
I’m not sure I understand the question. Just pick whichever one you prefer? You can’t have both a folder and a flat structure. But note that your sidebar hierarchy is orthogonal to where the pages live, so you can do something like this:

```js
export default {
  pages: [
    {
      name: "Section 1",
      path: "/section1",
      pages: [
        {name: "Page 1", path: "/page1"},
        {name: "Page 2", path: "/page2"}
      ]
    },
    {
      name: "Section 2",
      open: false,
      pages: [
        {name: "Page 3", path: "/page3"},
        {name: "Page 4", path: "/page4"}
      ]
    }
  ]
};
```

**Reply by Martien** [reply]
You are right. I was locked in tunnel vision, obsessed with getting this to work. The only way I can get what I want is flatten all pages – just as Wikipedia does.

Anyway, thanks for your reply and apologies for wasting your time.

Plus, have a nice weekend.
